File: backoffice/src/product/domain/events/ProductRenamed.py
"""
 *
 * Libraries 
 *
"""

from ..valueObjects.ProductName import ProductName
from ..valueObjects.ProductId   import ProductId
from src.shared.domain          import DomainEvent

"""
 *
 * Class
 *
"""

class ProductRenamed( DomainEvent ):

    """
     *
     * Attributes 
     *
    """

    __id   : ProductId
    __name : ProductName

    """
     *
     * Methods 
     *
    """

    def __init__( 
        self, 
        id        : ProductId   = None, 
        name      : ProductName = None,
        eventId   : str         = None,
        timestamp : int         = None
    ) -> None:
        super().__init__( eventId, timestamp )
        self.__id   = id
        self.__name = name
    
    def getId( self ) -> ProductId:
        return self.__id
    
    def getName( self ) -> ProductName:
        return self.__name
    
    def getEventName( self ) -> str:
        return 'product.renamed'
    
    def toPrimitives( self ) -> dict[str, str | int]:
        # Variables
        data : dict[str, str | int]
        # Code
        data = {
            'id'   : self.__id.getValue(),
            'name' : self.__name.getValue()
        }
        return data

    def fromPrimitives( self, 
        eventId   : str, 
        timestamp : int, 
        data      : dict[str, str | int]
    ) -> object:
        # Variables
        domainEvent : object
        # Code
        domainEvent = ProductRenamed(
            id        = ProductId( data['id'] ),
            name      = ProductName( data['name'] ),
            eventId   = eventId,
            timestamp = timestamp
        )
        return domainEvent

File: preparation/src/product/domain/events/ProductRenamed.cs
/* 
 * 
 * Libraries
 *
*/

using preparation.src.shared.domain;

/* 
 * 
 * Class
 *
*/

namespace preparation.src.product.domain {

    public class ProductRenamed : DomainEvent {

        /* 
         * 
         * Attributes
         *
        */

        private ProductId   id;
        private ProductName name;

        /* 
         * 
         * Methods
         *
        */

        public ProductRenamed() {
        }

        public ProductRenamed( ProductId id, ProductName name ) {
            this.id   = id;
            this.name = name;
        }

        public ProductRenamed(
            ProductId   id,
            ProductName name,
            string      eventId,
            int         timestamp
        ) : base( eventId, timestamp ) {
            this.id   = id;
            this.name = name;
        }

        public ProductId getId() {
            return id;
        }

        public ProductName getName() {
            return name;
        }

        public override DomainEvent fromPrimitives(
            string                     eventId, 
            int                        timestamp, 
            Dictionary<string, object> data
        ) {
            // Variables
            ProductRenamed domainEvent;
            // Code
            domainEvent = new ProductRenamed(
                new ProductId( data["id"].ToString() ),
                new ProductName( data["name"].ToString() ),
                eventId,
                timestamp
            );
            return domainEvent;
        }

        public override string getEventName() {
            return "product.renamed";
        }

        public override Dictionary<string, object> toPrimitives() {
            // Variables
            Dictionary<string, object> data;
            // Code
            data = new Dictionary<string, object>();
            data.Add( "id", id.getValue() );
            data.Add( "name", "" + name.getValue() );
            return data;
        }

    }

}